.github/workflows/docker.yml

```diff
@@ -0,0 +1,146 @@
+name: Build and test docker
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  get_dockerfiles:
+    name: Get list of dockerfiles for different containers
+    runs-on: ubuntu-latest
+    outputs:
+      docker_toml: ${{ steps.filter.outputs.docker_toml }}
+      dockerfiles: ${{ steps.set-matrix.outputs.dockerfiles }}
+      imagenames: ${{ steps.set-matrix.outputs.imagenames }}
+    steps:
+      - name: Checkout mrpro repo
+        uses: actions/checkout@v4
+
+      - name: Check if files in docker or the toml file has been modified
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            docker_toml:
+              - 'docker/*'
+              - 'pyproject.toml'
+              - '/.github/docker.yml'
+
+      - run: echo ${{ steps.filter.outputs.docker_toml }}
+
+      - id: set-matrix
+        if: steps.filter.outputs.docker_toml == 'true'
+        run: |
+          cd ./docker/
+          ls
```

Comment:
do you want to keep these debug statements?

Reply:
if yes, consider replacing 
` echo $dockerfiles `
by
` echo "dockerfiles=$dockerfiles"`
or print out a heading or something?

Reply:
i would say we can remove them...